oppia/oppia-android
# Fixes #1129: Code optimised for Backflow code

## Explanation


Fixes #1129

The `ExplorationActivity` when closed can go to different screens as per below mentioned scenarios:
1. When opened from `RecentlyPlayedActivity`, closing `ExplorationActivity` should open `TopicInfoTab`.
2. When opened from `StoryActivity`, closing `ExplorationActivity` should open `StoryActivity`.
3. When opened from `TopicLessonsTab`, closing `ExplorationActivity` should open `TopicLessonsTab`.

This implementation is working correctly in current code but it uses a term called `backflow` and it uses integer values to control various condition which makes it difficult to understand which screen will get opened just by looking at the code.

Therefore this PR introduces `ParentScreenForExplorationEnum` which is to make sure that the code becomes much more descriptive and understandable.

## How to Test:
Make sure that the app still fulfil these scenarios
1. When opened from `RecentlyPlayedActivity`, closing `ExplorationActivity` should open `TopicInfoTab`.
2. When opened from `StoryActivity`, closing `ExplorationActivity` should open `StoryActivity`.
3. When opened from `TopicLessonsTab`, closing `ExplorationActivity` should open `TopicLessonsTab`.

## Checklist
